packages/@lwc/engine-dom/src/apis/build-custom-element-constructor.ts

```diff
@@ -49,16 +50,26 @@ export function buildCustomElementConstructor(Ctor: ComponentConstructor): HTMLE
     const HtmlPrototype = getComponentHtmlPrototype(Ctor);
 
     return class extends HtmlPrototype {
+        isHydrated = false;
```

Comment:
In general, we try to limit the exposure of the LWC engine. Adding an `isHydrated` class field would expose this property on all the LWC native custom elements. I would recommend moving this to a `WeakSet` to avoid exposing it.

Reply:
I thought [WeakSet was unsupported in IE11](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WeakSet#browser_compatibility), but [WeakMap is supported](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WeakMap#browser_compatibility)? We have a comment elsewhere in the code about this:

https://github.com/salesforce/lwc/blob/4d463ffc2fc858ba7efbf6d08e7d4afde417e005/packages/%40lwc/engine-core/src/framework/rendering.ts#L474-L475

Reply:
Good point! I had forgotten about that. However, I don't think this code path applies to ie11. Also, IIRC, that comment/usage is because `WeakSet` is not supported natively, and (the important one) the polyfill is slow (`WeakSet` is polyfilled in compat).

I'll add a comment and leave this `WeakSet` usage unless someone has a strong opinion against it.